tunnel-server/src/proxy/index.ts

```diff
@@ -164,7 +166,7 @@ export const proxy = ({
       const { req: mutatedReq, activeTunnel } = await validateProxyRequest(
         tunnelFinder,
         req,
-        pkThumbprint => sessionStore(req, undefined as unknown as ServerResponse, pkThumbprint),
+        pkThumbprint => sessionStore(req, new ServerResponse(req), pkThumbprint),
```

Comment:
not sure creating a throwaway response makes a clearer statement than passing `undefined`. I'll try to refactor this later so it'll be possible to create a readonly `cookieSessionStore` but better to leave it out of this PR due to possible conflicts.

Reply:
I started with using a throw-away mock proxy based object, it worked, but it based on the awareness of the cookie's internal details doesn't manipulate the returned object from the Server request.
I thought it was a bit safer and cleaner that way, but I can also create the throw-away object inside the session store so it won't leak.

Reply:
Move it inside the session store.
Does it conflict with your PR?